chain/chunks/src/lib.rs

```diff
@@ -1366,7 +1365,7 @@ impl ShardsManager {
         );
 
         let block_producers =
-            self.runtime_adapter.get_epoch_block_producers_ordered(&epoch_id, &parent_hash)?;
+            self.runtime_adapter.get_epoch_block_approvers_ordered(&parent_hash)?;
```

Comment:
The test passes without this change. Keep it or revert?

Reply:
This change made the test pass before you made changes to the test instead? This seems like a good change to me; at the time I was writing this logic I was not thinking about the subtle difference between the producer and approver sets. But approvers is the right set to send to because the point of this logic is to help nodes tracking the shard to have enough parts to reconstruct the chunk without sending requests and I assume approvals will not be sent out until the chunks for shards they track are reconstructed.

Reply:
It changed from 0/10 pass to 2/10 pass, but need to change a bit more code there: `get_all_block_producers_ordered` uses a caching crutch that `get_all_block_approvers_ordered` doesn't have so it could increase cpu

Reply:
Yeah this change makes sense